main.nf

```diff
@@ -1,278 +1,2408 @@
+                          Check parameters
+================================================================================
+*/
+
+def json_schema = "$projectDir/nextflow_schema.json"
+if (params.help) {
+    def command = "nextflow run nf-core/circrna --input '*_R{1,2}.fastq.gz' -profile docker"
+    log.info NfcoreSchema.params_help(workflow, params, json_schema, command)
+    exit 0
+}
+
+if (params.validate_params) {
+    NfcoreSchema.validateParameters(params, json_schema, log)
+}
+
+// Check Tools selected
+toolList = defineToolList()
+tool = params.tool ? params.tool.split(',').collect{it.trim().toLowerCase()} : []
+if (!checkParameterList(tool, toolList)) exit 1, "[nf-core/circrna] error: Unknown tool selected, see --help for more information."
+
+// Check Modules selected
+moduleList = defineModuleList()
+module = params.module ? params.module.split(',').collect{it.trim().toLowerCase()} : []
+if (!checkParameterList(module, moduleList)) exit 1, "[nf-core/circrna] error: Unknown module selected, see --help for more information."
+
+// Check outdir not empty string.
+if(params.outdir == ''){
+   exit 1, "[nf-core/circrna] error: --outdir was not supplied, please provide a output directory to publish workflow results."
+}
+
+// Check input type not empty.
+if(params.input_type == ''){
+   exit 1, "[nf-core/circrna] error: --input_type was not supplied, please select 'fastq' or 'bam'."
+}
+
+// Check Genome version
+if(params.genome_version == ''){
+   exit 1, "[nf-core/circrna] error: --genome_version was not supplied, please select 'GRCh37' or 'GRCh38'."
+}
+
+// Check proper versions supplied
+if(params.genome_version){
+
+   GenomeVersions = defineGenomeVersions()
+
+   Channel
+         .value(params.genome_version)
+         .map{ it ->
+
+               if(!GenomeVersions.contains(it)){
+                  exit 1, "[nf-core/circrna] error: Incorrect genome version (${params.genome_version}) supplied.\n\nPlease select 'GRCh37' or 'GRCh38'"
+               }
+         }
+}
+
+/*
+ * The below Reference Genome / index parameters are allowed to be empty (they will be generated if empty)
+ * Mainly concerned about valid file extensions when provided.
+ */
+
+// Check Fasta
+if(params.fasta && !has_extension(params.fasta, ".fa") && !has_extension(params.fasta, ".fasta")){
+   exit 1, "[nf-core/circrna] error: Reference Fasta file provided (${params.fasta}) is not valid, Fasta file should have the extension '.fa' or '.fasta'."
+}
+
+// Check GTF
+if(params.gencode_gtf && !has_extension(params.gencode_gtf, ".gtf")){
+   exit 1, "[nf-core/circrna] error: Reference GTF file provided (${params.gencode_gtf}) is not valid, GTF file should have the extension '.gtf'."
+}
+
+// Check Fasta fai
+if(params.fasta_fai && !has_extension(params.fasta_fai, ".fai")){
+   exit 1, "[nf-core/circrna] error: Fasta index file provided (${params.fasta_fai}) is not valid, Fasta index files should have the extension '.fai'."
+}
```

Comment:
For the future it would be nice to abstract these away in a `Workflow.groovy` file in the `lib/´ folder.
(Check the `viralrecon` (dev) and/or `rnaseq` pipelines for examples)
But not necessary for this release, just thinking out loud :)